closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -474,6 +556,235 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+    // It is important to preserve the order!!!
+    static const char * const libNames[] = {
+
+    #if defined(_AIX)
+        "libcrypto.a(libcrypto64.so)",      // general symlink library name from archive file
+        "libcrypto64.so",                   // general symlink library name
+        "libcrypto.a(libcrypto.so)",        // general symlink library name from archive file
+        "libcrypto.so",                     // general symlink library name
+        "libcrypto.a(libcrypto64.so.3)",    // 3.x library name from archive file
+        "libcrypto64.so.3",                 // 3.x library nam
+        "libcrypto.a(libcrypto.so.3)",      // 3.x library name from archive file
+        "libcrypto.so.3",                   // 3.x library nam
+        "libcrypto.a(libcrypto64.so.1.1)",  // 1.1.x library name from archive file
+        "libcrypto.so.1.1",                 // 1.1.x library name
+        "libcrypto.so.1.0.0",               // 1.0.x library name
+    #elif defined(__APPLE__)
+        "libcrypto.3.dylib",                // 3.x library name
+        "libcrypto.1.1.dylib",              // 1.1.x library name
+        "libcrypto.1.0.0.dylib",            // 1.0.x library name
+        //"libcrypto.dylib"                 Apple no longer supports loading symlink.
+    #elif defined (_WIN32)
+        "libcrypto-3-x64.dll",              // 3.x library name
+        "libcrypto-1_1-x64.dll",            // 1.1.x library name
+        "libeay32.dll",                     // old library name
+    #else
+        "libcrypto.so",                     // general symlink library name
+        "libcrypto.so.3",                   // 3.x library name
+        "libcrypto.so.1.1",                 // 1.1.x library name
+        "libcrypto.so.1.0.0",               // 1.0.x library name
+        "libcrypto.so.10",                  // old library name
+    #endif
+    };
+
+    size_t size = (sizeof(libNames) / sizeof(libNames[0]));
+    // If JAVA_HOME is not null or empty and no library has been loaded yet, try there.
+    if ((NULL != chomepath) && strcmp(chomepath, "") && (NULL == crypto_library)) {
+        char **libNamesWithPath = malloc(size * sizeof(char *));
+        size_t path_len = strlen(chomepath);
+        char *libPath = malloc((path_len + 16) * sizeof(char));
+        strcpy(libPath, chomepath);
+        // Append the proper directory using a slash or backslash, depending on the operating system.
+        #if defined(_WIN32)
+        strcat(libPath, "\\bin\\");
+        #else
+        strcat(libPath, "/lib/");
+        #endif
+        if (traceEnabled) {
+            fprintf(stdout, "Attempting to load library bundled with JDK from: %s\n", libPath);
+        }
+
+        for (int i = 0; i < size; i++) {
+            size_t file_len = strlen(libNames[i]);
+            // Allocate memory for the new file name with the path.
+            libNamesWithPath[i] = malloc((path_len + file_len + 16) * sizeof(char));
```

Comment:
There's no need to multiply by `sizeof(char)` which is guaranteed to be `1`; missing cast.
```c
            libNamesWithPath[i] = (char *)malloc(path_len + file_len + 16);
```
This is missing a check that `malloc()` succeeds.

Reply:
Removed `sizeof(char)`, added cast and check.